tensorflow/python/summary/summary_iterator.py

```diff
@@ -301,26 +306,35 @@ def __init__(self, queue, ev_writer, flush_secs):
     """
     threading.Thread.__init__(self)
     self.daemon = True
+    self._stop_flag = False
     self._queue = queue
     self._ev_writer = ev_writer
     self._flush_secs = flush_secs
     # The first event will be flushed immediately.
     self._next_event_flush_time = 0
+    self._fetch_timeout = 1
 
   def run(self):
     while True:
-      event = self._queue.get()
+      event = self._queue.get(timeout=self._fetch_timeout)
       try:
-        self._ev_writer.WriteEvent(event)
+        if event:
+          self._ev_writer.WriteEvent(event)
         # Flush the event writer every so often.
         now = time.time()
-        if now > self._next_event_flush_time:
+        if now > self._next_event_flush_time or self._stop_flag:
```

Comment:
Is it necessary to flush when `self._stop_flag` is true? The `stop()` method also calls `flush()`.